src/SW_netCDF.c

```diff
@@ -1896,16 +2265,25 @@ void SW_NC_deconstruct(SW_NETCDF* SW_netCDF) {
  *  netCDF file information
 */
 void SW_NC_open_files(SW_NETCDF* SW_netCDF, LOG_INFO* LogInfo) {
-    int fileNum;
+    int fileNum, openType, *fileID;
+    char* fileName;
 
     for(fileNum = 0; fileNum < SW_NVARNC; fileNum++) {
-        if(FileExists(SW_netCDF->InFilesNC[fileNum])) {
-            if(nc_open(SW_netCDF->InFilesNC[fileNum], NC_NOWRITE,
-                                    &SW_netCDF->ncFileIDs[fileNum]) != NC_NOERR) {
+        fileName = SW_netCDF->InFilesNC[fileNum];
+        fileID = &SW_netCDF->ncFileIDs[fileNum];
+
+        if(FileExists(fileName)) {
+            openType = (fileNum == vNCprog) ? NC_WRITE : NC_NOWRITE;
 
+            if(nc_open(fileName, openType, fileID) != NC_NOERR) {
```

Comment:
How cumbersome would it be to enable a user that wants the "progress" variable to be added to the "domain.nc"?

Currently, if I change "files_nc.in" to `progress progress Input_nc/domain.nc`, then `nc_open(fileName, openType, fileID)` fails with error code NC_EHDFERR (-101) for "progress".

Reply:
This is a good point. The error is most likely coming from the fact that `domain.nc` would be opened in more than one mode `NC_NOWRITE` and `NC_WRITE`. So to answer the question, it would depend on how it is implemented.

* Most likely, the best way would be to just open all input files with purpose to write to it. Then we modify the testing of the template/progress variable and add the variable if needed. And since this most likely opens up two file handlers, at the end we could close the progress ID and swap it with the domain ID if they are the same.

There are most likely other ways of handling this, but I'd say we discuss those if needed.

Reply:
This works nicely, thanks!